### DISCUSSION TITLE
[优化建议] 自动更新检测或许可以考虑使用 GitHub releases api?

### DISCUSSION THREAD

**Discussion by ak1ra-komj** [wushuo894/ani-rss#128] [Ideas]
好奇 ani-rss 怎么实现的 自动更新, 简单翻了下相关代码,

https://github.com/wushuo894/ani-rss/blob/cd896680c7cc8716ef0ae5ac3dd86651cc64fbe8/src/main/java/ani/rss/util/UpdateUtil.java#L35-L67

看起来好像在直接在处理 releases 页面的 HTML, 感觉这部分逻辑或许可以考虑使用 GitHub releases api? 返回的 JSON 格式数据应该更方便处理,

https://api.github.com/repos/wushuo894/ani-rss/releases/latest

不过我不会写 Java 项目, 只能看懂一些简单的逻辑, 否则我就直接提 PR 了, 这个建议优先级也不是很高, 按照你原本的节奏安排就好.

**Comment by wushuo894** [comment]
因为直接爬网页可以获取到html，github会把图片或者@.#等md做完了处理，这样我可以在更新弹窗里直接显示图片与问题、@了,也不需要自己再处理一遍md格式。api接口里面还是原始md的文本

**Reply by ak1ra-komj** [reply]
哦哦, 再看了下 GitHub Actions, releases body 来自于 UPDATE.md 这个 Markdown 文件, releases 的 HTML 页面的 body 已经经过 GitHub 渲染, 抓 HTML 就可以直接把这部分内容填到弹窗的正文中而不用自己再处理 Markdown 了;
